File: Python/lesson 01/BinaryGap.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(N):
    # write your code in Python 3.6
    result = 0
    counter = 0
    is_start = False
    while N != 0:
        bit = N % 2
        N = N >> 1
        if bit:
            if is_start:
                result = counter if counter > result else result
                counter = 0
            else:
                is_start = True
        elif is_start:
            counter += 1
    return result

File: JavaScript/lesson 01/BinaryGap.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(N) {
    // write your code in JavaScript (Node.js 8.9.4)
    var result = 0;
    var counter = 0;
    var isStarted = false;
    while (N != 0) {
        var bit = N % 2;
        N = N >> 1;
        if (bit) {
            if (isStarted) {
                if (result < counter) {
                    result = counter;
                }
                counter = 0;
            } else {
                isStarted = true;
            }
        } else if (isStarted) {
            counter++;
        }
    }
    return result;
}